frontend/class-woocommerce-shop-page.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Represents the logic to determine if the current page is a WooCommerce shop page.
+ */
+class WPSEO_WooCommerce_Shop_Page {
+
+	/**
+	 * Checks if the current page is the shop page..
+	 *
+	 * @return bool True when the current page is the shop page.
+	 */
+	public function is_shop_page() {
+		return $this->is_current_shop_page() && $this->get_page_id() !== -1;
```

Comment:
Can you explain why `is_current_shop_page` is not enough? From first glance `is_shop() && ! is_search()` should fully determine if the current page is a WooCommerce shop page. If it doesn't, then that needs to be explained in a comment.